scripts/pylib/pytest-twister-harness/src/twister_harness/helpers/babblesim.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import logging
+import shutil
+import subprocess
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class BabbleSim:
+    def __init__(self):
+        self.process: subprocess.Popen | None = None
+
+    @staticmethod
+    def copy_app(bsim_env: dict, build_dir: Path, new_app_name: str):
+        new_app_path = Path(bsim_env['BSIM_OUT_PATH']) / 'bin' / new_app_name
+        original_app_path = build_dir / 'zephyr' / 'zephyr.exe'
+        logger.debug(f'Copy application {original_app_path} to {new_app_path}')
+        shutil.copy(original_app_path, new_app_path)
```

Comment:
You are right – this a huge simplification on my side and of course it can’t work in this way. The root cause is that I fixed the name of copied binary application without taking into account that this name can vary from the used platform (`nrf52_bsim` or  `nrf5340bsim_nrf5340_cpunet`):
```python
def test_adv_chain(bsim_env: dict, build_dir: Path):
    new_app_name = 'bs_nrf52_bsim_tests_bsim_bluetooth_host_adv_chain_prj_conf'
```
This should be fixed by one of following ideas:
1.	Pass to pytest test information about used platform name and then set the `new_app_name` basing on this information. This requires also to set `BOARD` env variable to pass this info to executed bash script.
2.	Generate `new_app_name` variable basing on build directory path (this could guarantee that new binary names will be unique). This solution requires also changes in bash scripts – they should “know” what is `new_app_name` to make it possible to run proper (and existing) executable.
```python
def copy_app(bsim_env: dict, build_dir: Path):
	zephyr_base_path = Path(bsim_env['ZEPHYR_BASE'])
	if zephyr_base_path in build_dir.parents:
		new_app_name = build_dir.relative_to(zephyr_base_path)
	else:
		new_app_name = build_dir
	new_app_name = re.sub(r'[/\\.-]', '_', str(new_app_name))
	new_app_path = Path(bsim_env['BSIM_OUT_PATH']) / 'bin' / new_app_name
```
3.	Resign from coping binaries (as was mentioned in first comment)